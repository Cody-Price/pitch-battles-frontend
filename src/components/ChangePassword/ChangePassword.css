.change-password-form {
  display: grid;
  grid-template-columns: 130px 1fr;
  animation: account-subsection-fade-in 0.3s linear;
}

.change-password-form *:focus {
  outline: none;
}

.change-password-form button:active {
  box-shadow: none;
}

.change-password-form button {
  grid-column: 1/3;
  width: 150px;
  height: 23px;
  font-family: "chronotype";
  font-size: 1rem;
  margin: 1rem auto;
  background: #6a9dd4;
  border: 0;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.7);
}

.change-password-form input {
  font-family: "chronotype";
  font-size: 1rem;
  height: 18px;
  width: 300px;
  margin-top: 1rem;
  background: #6a9dd4;
  border: 0;
  padding-left: 8px;
}

.change-password-form input:placeholder {
  color: #333;
}

.change-password-form p {
  padding-top: 1.3rem;
  margin: 0;
}
