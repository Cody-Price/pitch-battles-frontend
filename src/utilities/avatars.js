const avatars = [
  "one",
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight",
  "nine",
  "ten",
  "eleven",
  "twelve",
  "thirteen",
  "fourteen"
];

export default avatars;
