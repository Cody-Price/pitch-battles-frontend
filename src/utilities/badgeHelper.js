export const badgeHelper = [
  { name: "level_one", badge: "level one: completed", rank: 2 },
  { name: "level_one_perfect", badge: "level one: perfect", rank: 3 },
  { name: "level_two", badge: "level two: completed", rank: 4 },
  { name: "level_two_perfect", badge: "level two: perfect", rank: 6 },
  { name: "level_three", badge: "level three: completed", rank: 7 },
  { name: "level_three_perfect", badge: "level three: perfect", rank: 9 },
  { name: "level_four", badge: "level four: completed", rank: 11 },
  { name: "level_four_perfect", badge: "level four: perfect", rank: 12 },
  { name: "all_perfect", badge: "all levels: perfect", rank: 14 },
  { name: "five_games", badge: "play 5 games", rank: 1 },
  { name: "ten_games", badge: "play 10 games", rank: 5 },
  { name: "twenty_games", badge: "play 20 games", rank: 8 },
  { name: "fifty_games", badge: "play 50 games", rank: 10 },
  { name: "one_hundred_games", badge: "play 100 games", rank: 13 },
  { name: "two_hundred_games", badge: "play 200 games", rank: 15 },
  { name: "five_hundred_games", badge: "play 500 games", rank: 16 },
  { name: "one_thousand_games", badge: "play 1000 games", rank: 17 }
];
