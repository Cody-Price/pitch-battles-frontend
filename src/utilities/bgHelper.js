export const bgHelper = [
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight"
];
