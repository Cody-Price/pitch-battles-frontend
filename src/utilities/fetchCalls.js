export const login = async body => {
  const url = "https://pitch-battles-api.herokuapp.com/login";

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };

  const response = await fetch(url, options);
  return await response.json();
};

export const signUp = async body => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/users";

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };

  const response = await fetch(url, options);
  return await response.json();
};

export const userFetch = async webToken => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/dashboard/";

  const options = {
    method: "GET",
    mode: "cors",
    headers: {
      Authorization: `bearer ${webToken}`,
      "Content-Type": "application/json"
    }
  };

  console.log(options);

  const response = await fetch(url, options);
  return await response.json();
};

export const postGameUserUpdate = async (update, webToken) => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/users";

  const body = update;

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };
  return;
};

export const forgotMyPasswordCall = async email => {
  // const url
  // return response
  return Promise.resolve(true);
};
