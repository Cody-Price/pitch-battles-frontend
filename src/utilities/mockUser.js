const mockUser = {
  first_name: "Chet",
  last_name: "Manly",
  avatar: 10,
  role: 0,
  email: "cmanly@gmail.com",
  games_played: 100,
  class: "6th-grade band"
};

export default mockUser;
