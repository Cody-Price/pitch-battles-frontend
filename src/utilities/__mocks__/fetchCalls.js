import mockUser from "../../utilities/mockUser";
import mockStudentClass from "../mockStudentClass";
import { mockTeacherClass } from "../mockTeacherClass";

export const login = async body => {
  const mockResponse = { access_token: "test" };
  return Promise.resolve(mockResponse);
};

export const signUp = async body => {
  const mockResponse = { response: {} };

  return Promise.resolve(mockResponse);
};

export const userFetch = async webToken => {
  const mockResponse = mockUser;
  return Promise.resolve(mockResponse);
};

export const postGameUserUpdate = async (update, webToken) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const forgotMyPasswordCall = async email => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const changeAvatarFetch = async (avatar, id, webToken) => {
  const mockResponse = { reponse: [] };

  return Promise.resolve(mockResponse);
};

export const changeProfileFetch = async (name, id, webToken) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const changePasswordFetch = async (
  oldPassword,
  newPassword,
  id,
  webToken
) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const createClassFetch = async (name, webToken) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const joinClassFetch = async (class_key, id, webToken) => {};

export const leaveClassFetch = async (student_id, class_id, webToken) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const resetPasswordFetch = async (password, token) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const studentClassFetch = async webToken => {
  const mockResponse = mockStudentClass;

  return Promise.resolve(mockResponse);
};

export const teacherAllClassesFetch = async webToken => {
  const mockResponse = mockTeacherClass;

  return Promise.resolve(mockResponse);
};

export const teacherSpecificClassFetch = async (id, webToken) => {
  const mockResponse = mockTeacherClass;

  return Promise.resolve(mockResponse);
};

export const deleteClassFetch = async (id, webToken) => {
  const mockResponse = { data: "success" };

  return Promise.resolve(mockResponse);
};
