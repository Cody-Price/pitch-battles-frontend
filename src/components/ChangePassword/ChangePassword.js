import React, { Component } from "react";
import { login } from "../../utilities/fetchCalls";
import "./ChangePassword.css";

class ChangePassword extends Component {
  constructor() {
    super();

    this.state = {
      oldPassword: "",
      newPassword: "",
      confirmPassword: "",
      incompleteError: false,
      incorrectPasswordError: false,
      passwordMatchError: false
    };
  }

  handleSubmit = async event => {
    event.preventDefault();
    if (
      !this.state.oldPassword ||
      !this.state.newPassword ||
      !this.state.confirmPassword
    ) {
      this.setState({
        incompleteError: true
      });
    }

    if (this.state.newPassword !== this.state.confirmPassword) {
      this.setState({
        passwordMatchError: true
      });
      return;
    }

    const passwordConfirmed = await this.confirmOldPassword();
    if (!passwordConfirmed) {
      return;
    }

    this.props.updatePassword(this.state.newPassword);
  };

  confirmOldPassword = async () => {
    const body = {
      email: this.props.user.email,
      password: this.state.oldPassword
    };

    try {
      const response = await login(body);
      if (response.access_token) {
        this.props.updateWebToken(response.access_token);
        return true;
      }
    } catch (error) {
      console.log(error);
      this.setState({
        incorrectPasswordError: true
      });
      return false;
    }

    // this.props.updateWebToken(body.)
  };

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  render() {
    return (
      <section className={`change-password`}>
        <form
          className="change-password-form"
          onSubmit={event => {
            this.handleSubmit(event);
          }}
        >
          <p className="password-component-text">old password</p>
          <input
            type="password"
            name="oldPassword"
            value={this.state.oldPassword}
            placeholder="old password"
            onChange={event => this.handleChange(event)}
            className="old-password-input"
          />
          <p className="password-component-text">new password</p>
          <input
            type="password"
            name="newPassword"
            value={this.state.newPassword}
            placeholder="new password"
            onChange={event => this.handleChange(event)}
            className="new-password-input"
          />

          <p className="password-component-text">confirm password</p>
          <input
            type="password"
            name="confirmPassword"
            value={this.state.confirmPassword}
            placeholder="confirm password"
            onChange={event => this.handleChange(event)}
            className="new-password-confirm-input"
          />

          <button type="submit">change password</button>
        </form>
      </section>
    );
  }
}

export default ChangePassword;
