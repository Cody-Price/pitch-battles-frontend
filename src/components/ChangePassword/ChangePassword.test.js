import React from "react";
import * as fetchCalls from "../../utilities/fetchCalls";
import mockUser from "../../utilities/mockUser";

import ChangePassword from "./ChangePassword";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("ChangePassword", () => {
  let wrapper;
  let mockUpdateWebToken;

  beforeEach(() => {
    mockUpdateWebToken = jest.fn();
    wrapper = shallow(
      <ChangePassword updateWebToken={mockUpdateWebToken} user={mockUser} />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleSubmit", () => {
    let mockPreventDefault;

    it("should call preventDefault", () => {
      mockPreventDefault = jest.fn();

      const mockEvent = {
        target: {
          name: "oldPassword",
          value: "I am an old password"
        },
        preventDefault: mockPreventDefault
      };

      wrapper.instance().handleSubmit(mockEvent);

      expect(mockPreventDefault).toHaveBeenCalled();
    });

    it("should set an error if fields are incomplete", () => {
      mockPreventDefault = jest.fn();

      const mockEvent = {
        target: {
          name: "oldPassword",
          value: "I am an old password"
        },
        preventDefault: mockPreventDefault
      };
      const expected = true;

      wrapper.instance().handleSubmit(mockEvent);

      expect(wrapper.state().incompleteError).toEqual(expected);
    });

    it("should set an error if the new passwords do not match", () => {
      mockPreventDefault = jest.fn();

      const mockEvent = {
        target: {
          name: "oldPassword",
          value: "I am an old password"
        },
        preventDefault: mockPreventDefault
      };
      const expected = true;

      wrapper.setState({
        newPassword: "Hello",
        confirmPassword: "Goodbye",
        oldPassword: "What"
      });

      wrapper.instance().handleSubmit(mockEvent);

      expect(wrapper.state().passwordMatchError).toEqual(true);
    });

    it("should call confirmOldPassword", async () => {
      const mockConfirmOldPassword = jest.fn().mockImplementation(() => {
        return Promise.resolve();
      });

      mockPreventDefault = jest.fn();

      const mockEvent = {
        target: {
          name: "oldPassword",
          value: "I am an old password"
        },
        preventDefault: mockPreventDefault
      };

      wrapper.setState({
        newPassword: "Hello",
        confirmPassword: "Hello",
        oldPassword: "What"
      });

      wrapper.instance().confirmOldPassword = mockConfirmOldPassword;

      await wrapper.instance().handleSubmit(mockEvent);

      expect(mockConfirmOldPassword).toHaveBeenCalled();
    });

    it("should return if the password is not confirmed", async () => {
      const mockConfirmOldPassword = jest.fn().mockImplementation(() => {
        return Promise.resolve(false);
      });

      mockPreventDefault = jest.fn();

      const mockEvent = {
        target: {
          name: "oldPassword",
          value: "I am an old password"
        },
        preventDefault: mockPreventDefault
      };

      const spy = jest.spyOn(wrapper.instance(), "handleSubmit");

      wrapper.setState({
        newPassword: "Hello",
        confirmPassword: "Hello",
        oldPassword: "What"
      });

      wrapper.instance().confirmOldPassword = mockConfirmOldPassword;

      await wrapper.instance().handleSubmit(mockEvent);

      expect(spy).toReturn();
    });

    it("should call handleSubmit on submit", () => {
      mockPreventDefault = jest.fn();

      const mockEvent = {
        target: {
          name: "oldPassword",
          value: "I am an old password"
        },
        preventDefault: mockPreventDefault
      };
      const spy = jest.spyOn(wrapper.instance(), "handleSubmit");
      wrapper.instance().forceUpdate();

      wrapper.find(".change-password-form").simulate("submit", mockEvent);

      expect(spy).toHaveBeenCalled();
    });
  });

  describe("confirmOldPassword", async () => {
    it("should call login with the correct params", async () => {
      const expected = {
        email: "simpsonkevinjohn@gmail.com",
        password: "What"
      };

      wrapper.setState({
        newPassword: "Hello",
        confirmPassword: "Hello",
        oldPassword: "What"
      });

      fetchCalls.login = jest.fn().mockImplementation(() => {
        return Promise.resolve({ access_token: "string" });
      });

      await wrapper.instance().confirmOldPassword();

      expect(fetchCalls.login).toBeCalledWith(expected);
    });

    it("should return true if login resolves", async () => {
      wrapper.setState({
        newPassword: "Hello",
        confirmPassword: "Hello",
        oldPassword: "What"
      });

      fetchCalls.login = jest.fn().mockImplementation(() => {
        return Promise.resolve({ access_token: "string" });
      });

      const resolve = await wrapper.instance().confirmOldPassword();

      expect(resolve).toEqual(true);
    });

    it("should should call updateWebToken with the correct params if login resolves", async () => {
      wrapper.setState({
        newPassword: "Hello",
        confirmPassword: "Hello",
        oldPassword: "What"
      });

      fetchCalls.login = jest.fn().mockImplementation(() => {
        return Promise.resolve({ access_token: "string" });
      });

      const expected = "string";

      await wrapper.instance().confirmOldPassword();

      expect(mockUpdateWebToken).toBeCalledWith(expected);
    });

    it("should setState if login throws an error", async () => {
      wrapper.setState({
        newPassword: "Hello",
        confirmPassword: "Hello",
        oldPassword: "What"
      });

      fetchCalls.login = jest.fn().mockImplementation(() => {
        return Promise.reject({ message: "haha" });
      });

      const expected = true;

      await wrapper.instance().confirmOldPassword();

      expect(wrapper.state().incorrectPasswordError).toEqual(true);
    });

    it("should return false if login throws an error", async () => {
      wrapper.setState({
        newPassword: "Hello",
        confirmPassword: "Hello",
        oldPassword: "What"
      });

      fetchCalls.login = jest.fn().mockImplementation(() => {
        return Promise.reject({ access_token: "hahaha" });
      });

      const response = await wrapper.instance().confirmOldPassword();

      expect(response).toEqual(false);
    });
  });

  describe("handleChange", () => {
    it("should setState", () => {
      const expected = "I am an old password";
      const mockEvent = {
        target: {
          name: "oldPassword",
          value: "I am an old password",
          preventDefault: jest.fn()
        }
      };

      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state().oldPassword).toEqual(expected);
    });

    it("should call handleChange on oldPassword Change with the correct params", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleChange");

      const mockEvent = {
        target: {
          name: "oldPassword",
          value: "I am an old password",
          preventDefault: jest.fn()
        }
      };

      wrapper.find(".old-password-input").simulate("change", mockEvent);

      expect(spy).toBeCalledWith(mockEvent);
    });

    it("should call handleChange on newPassword Change with the correct params", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleChange");

      const mockEvent = {
        target: {
          name: "newPassword",
          value: "I am an new password",
          preventDefault: jest.fn()
        }
      };

      wrapper.find(".new-password-input").simulate("change", mockEvent);

      expect(spy).toBeCalledWith(mockEvent);
    });

    it("should call handleChange on confirmPassword Change with the correct params", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleChange");

      const mockEvent = {
        target: {
          name: "confirmPassword",
          value: "I am an new password",
          preventDefault: jest.fn()
        }
      };

      wrapper.find(".new-password-confirm-input").simulate("change", mockEvent);

      expect(spy).toBeCalledWith(mockEvent);
    });
  });

  describe("changePassword", () => {
    it("should setState", async () => {
      await wrapper.instance().changePassword();

      console.log(wrapper.state().incompleteError);
    });
  });
});
