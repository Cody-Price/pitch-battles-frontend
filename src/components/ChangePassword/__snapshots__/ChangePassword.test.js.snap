// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword should match the snapshot 1`] = `
<section
  className="change-password"
>
  <form
    className="change-password-form"
    onSubmit={[Function]}
  >
    <p
      className="password-component-text"
    >
      old password
    </p>
    <input
      className="old-password-input"
      name="oldPassword"
      onChange={[Function]}
      placeholder="old password"
      type="password"
      value=""
    />
    <p
      className="password-component-text"
    >
      new password
    </p>
    <input
      className="new-password-input"
      name="newPassword"
      onChange={[Function]}
      placeholder="new password"
      type="password"
      value=""
    />
    <p
      className="password-component-text"
    >
      confirm password
    </p>
    <input
      className="new-password-confirm-input"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="confirm password"
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      change password
    </button>
  </form>
</section>
`;
